### DISCUSSION TITLE
Ability to auto-change directory based on project root

### DISCUSSION THREAD

**Discussion by leelavg** [helix-editor/helix#5123] [Q&A]
Hi,

- I skimmed through some of the existing discussions and issues, however couldn't find the feature/config being asked and so raising a new Q&A
- Can below be possible using helix-editor (I'm using some of Vim's terminology & I believe it's same here as well)?
1. Assume I'm at any dir in a project, let's say `proj-a` and currently at `~/proj-a/dir1/file2`, `~/proj-a` contains `.git` folder
2. Can I auto-set (or by any keymap) the current working directory as `~/proj-a`
3. Now I open a file in another window from another project `proj-b` and currently at `~/proj-b/dir3/file4` and now in that buffer can the cwd be set as `~/proj-b`?
- This is the main (or only) functionality that I'm missing from Vim for daily use, any help would be appreciated. Thanks.
- Below is how it's currently used in Neovim
https://github.com/leelavg/dotfiles/blob/ec6934215fb822e1d914ffa7eef9cfccd790e8f8/config/init.vim#L75-L102
https://github.com/leelavg/dotfiles/blob/ec6934215fb822e1d914ffa7eef9cfccd790e8f8/config/init.vim#L269

**Comment by Syntonica** [comment]
I second this: either automatic, or a command to do so.

**Comment by WuerfelDev** [comment]
Here's a workaround to run a command from the project root. But I don't think we can use that to change the pwd inside of helix automatically.
First we need to set up a git alias 'exec' (which can execute any command at its project root): `git config --global alias.exec '!exec '`
Exemplary usage is like this: `git exec make` to run the make command

Now in the local _<projectfolder>/.helix/config.toml_ we can create keybinds with it:
```
[keys.normal."+"]
b = ":run-shell-command git exec make"
c = ":run-shell-command git exec make clean"
```
